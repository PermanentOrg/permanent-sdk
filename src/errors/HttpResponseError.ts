export class HttpResponseError extends Error {
  public readonly statusCode: number;

  public constructor(statusCode: number, message: string) {
    super(message);
    this.statusCode = statusCode;
  }
}
