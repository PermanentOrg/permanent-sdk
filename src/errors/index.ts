export * from './HttpResponseError';
export * from './ValidationError';
