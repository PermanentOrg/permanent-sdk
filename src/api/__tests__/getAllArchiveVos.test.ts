import path from "node:path";
import nock from "nock";
import { getAllArchiveVos } from "..";
import { ValidationError } from "../../errors";

describe("getAllArchiveVos", () => {
	it("should return an array of archive VOs", async () => {
		nock("https://permanent.local")
			.get("/api/archive/getAllArchives")
			.replyWithFile(
				200,
				path.join(
					__dirname,
					"fixtures",
					"getAllArchiveVos",
					"multipleArchives.json",
				),
				{
					"Content-Type": "application/json",
				},
			);

		const archiveVos = await getAllArchiveVos({
			bearerToken: "12345",
			baseUrl: "https://permanent.local/api",
		});

		expect(archiveVos).toMatchSnapshot();
	});

	it("should error when provided an invalid response", async () => {
		nock("https://permanent.local")
			.get("/api/archive/getAllArchives")
			.reply(200, '["Not at all an archiveVO"]', {
				"Content-Type": "application/json",
			});

		await expect(
			getAllArchiveVos({
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			}),
		).rejects.toBeInstanceOf(ValidationError);
	});
});
