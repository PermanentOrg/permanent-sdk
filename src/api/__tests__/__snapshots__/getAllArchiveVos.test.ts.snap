// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllArchiveVos should return an array of archive VOs 1`] = `
[
  {
    "ChildFolderVOs": [],
    "FolderSizeVOs": [],
    "ItemVOs": [],
    "RecordVOs": [],
    "accessRole": "access.role.owner",
    "allowPublicDownload": 1,
    "archiveId": 1,
    "archiveNbr": "0001-0000",
    "birthDay": null,
    "company": null,
    "createdDT": "2022-06-23T18:33:43",
    "description": null,
    "fileLeft": null,
    "fileTotal": null,
    "fullName": "dan",
    "homeCity": null,
    "homeCountry": null,
    "homeState": null,
    "imageRatio": "1.00",
    "public": null,
    "publicDT": "2022-06-23T18:33:43",
    "relationType": null,
    "spaceLeft": null,
    "spaceTotal": null,
    "status": "status.generic.ok",
    "thumbArchiveNbr": null,
    "thumbDT": "2023-06-23T18:33:57",
    "thumbStatus": "status.generic.ok",
    "thumbURL1000": "https://localcdn.permanent.org/_slifty/0001-0000.thumb.w1000?t=1687545237&Expires=1687545237&Signature=PS54XhzqHjNU3m1Ct2QasVoxlt92mzS0~t0Wg1GpqddEcP0juL1mgo0ap-gvmjhhiTDd6SUghBhVExUvGewZ9UGclqbEm13y~VPKBh5VbqrzOzeKA1astazF9bKDYX1WUxetU-M6C1ES~pNjOT5nTsqpgEqNtRM7LGxqHix9LxzTYXLcyJl9Vh1v9eAwn64W2DbH9ORJXlq5R09uyfKaLlDFY~2tKiiONFJlSMreiDfYCfLixWrKuiTepxkVqmf59uFx-xiVzOoGpGzDNl3U0~l0WYSXFUWfl-9WeS3Az1TnzC9OyAuk5imWBlT70P1MgjAcKyvM6EhpWV~hjZakjA__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "thumbURL200": "https://localcdn.permanent.org/_slifty/0001-0000.thumb.w200?t=1687545237&Expires=1687545237&Signature=RyBFbgZNpy9wP9ouSHGm~JBqas7tVO4d2eVrKP4cO99kUCGBfnnL5niEge53Hx8mULrHJbXZp-Yfk92B4DS8Mfl8aAtz9EIGZlQcOUa6Q9znGEtmk54vVhbHg8afgifwacCxVec3FC5WhTH6p-N8lXu~sfdYitSmuvn8VFMBMVvGFd5E63CGg639tfXnGgLDhYNZrkMnFKt8-bIPJenPhV1P7JatrLR2vs1kuAw8y-F9EWodJOeOrQCWcN9o1ID1KaQ308jdjj42fWvwzQXXfZkUvEDKg7znsHzbAGz7e~c0T6mMwHXJT-pYnhqWK0wgWmmavUxXp2i5D2R1W~qHnA__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "thumbURL2000": "https://localcdn.permanent.org/_slifty/0001-0000.thumb.w2000?t=1687545237&Expires=1687545237&Signature=ImVXZcuCpts5afm8BY7nSxgGa0shT0auBweG8sZlkVO2DSn0v9xM~CuDC4TexQCl80h9ZDrtszftbFXyQuE-k0BsGq8G2632EVLTbE~qcVpn9RVEeUvuzV2buGMdWb3LO6hauMjsiv1~aDUAuv5iIEY0ki9CAkBp2NX6ZReAPja2Iru8ECUK8SMFg7-p-WGUQmkgTj9Eil1wRP5~V2~j6v4RhmG0NqhEYcYn1SK02zeh4Ii~j-FpVEnfvy1BJ5a8SLFvuDRSTMTf~-G0G0YPdtiPkloiPFlWqKt19pMmWrPL~r9Xha2Rbl-MrYu2TLVIkUEYfx43RijrCrKiLLTBXg__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "thumbURL500": "https://localcdn.permanent.org/_slifty/0001-0000.thumb.w500?t=1687545237&Expires=1687545237&Signature=IqGadDfOEWeeO0KqN9G1zSDkOEPdQcq9CV2e3VN8D69WWzm1Ya139Y3e8HV6WYgwc9bXnF7E4tt2vw8DOWfcHptNUbnLuK7DezftcFECip7gW0oDmnO0anWJ3y5EoxqmOCJyZQQQVKRKs736psOqs2zk~npX~UaIak-o1gOR6QUwV3T0QoAH2NHLGiqYa2xmecjt2VVt2-UB58FSVgwDKO~R4hIqVA2NIdqAWMKVXAHfDVjNa9T8Gh0pnH6BgzsF7rq-5kzlHfFnJqUXwQejoR8fqRmoMLiL-pdVJog2bIb6HIdEWUU0B1P8FQtFDI~bEXnB4ijHk6GOpg-CEHug9A__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "type": "type.archive.person",
    "updatedDT": "2022-06-23T18:33:57",
    "view": null,
    "viewProperty": null,
  },
  {
    "ChildFolderVOs": [],
    "FolderSizeVOs": [],
    "ItemVOs": [],
    "RecordVOs": [],
    "accessRole": "access.role.owner",
    "allowPublicDownload": 1,
    "archiveId": 20,
    "archiveNbr": "000k-0000",
    "birthDay": null,
    "company": null,
    "createdDT": "2022-06-23T18:34:25",
    "description": null,
    "fileLeft": null,
    "fileTotal": null,
    "fullName": "another test account",
    "homeCity": null,
    "homeCountry": null,
    "homeState": null,
    "imageRatio": "1.00",
    "public": null,
    "publicDT": "2022-06-23T18:34:25",
    "relationType": null,
    "spaceLeft": null,
    "spaceTotal": null,
    "status": "status.generic.ok",
    "thumbArchiveNbr": null,
    "thumbDT": "2023-06-23T19:38:01",
    "thumbStatus": "status.archive.gen_avatar",
    "thumbURL1000": "https://localcdn.permanent.org/_slifty/000k-0000.thumb.w1000?t=1687549081&Expires=1687549081&Signature=X-XPDjHlpD79NsvoAGuAF1rc56e3b4BIH1hERXHAHKpnmfKn35~BgmjvlHNfOu2yBLF7A1VgoWFWDDvDqclIbnpUbFcISBg3goh4c4~CS~Om8a6Y~LsoS~GOgbRxu~dR5hclHdeULcZSI70BGg4mGUrQYyjONaBtaQPoE51cLrPVjk4dYnF84zMhLLTzK09WQVK6zmxw2~84pWjSySyf~zVRHMS4KPZ8xg6KhI9VTkdIHXsuruRSjSVYiV5tVWoxD1AWYGy0QTiBUhjrbZYxkQN5Lz4pDb6ovOn7qOob0~4C~rTl6cMwxb9iwJRtLFCj~m4Eu8jXxYP5lXcZ-G3P1A__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "thumbURL200": "https://localcdn.permanent.org/_slifty/000k-0000.thumb.w200?t=1687549081&Expires=1687549081&Signature=E~35Yh6c7beK6HFG6ZnmFjANvYK-7NL-JoFHePnbGGFZpExhk5p7m7eIhaB76~5lkL1ZqCNEyJwL0dR-c2P-5SMXcjTk7GJxDRkSv9D9GW0sMje3EGGnXEVX8G0xri3Fn4SPLANUVXEpWC1uWoCNA8qTxIoS9VyAWzWQ7LJQI-ZA1-~8UkScM-5uhPIjaOSgcmA3SKAMuqcWiRtvE7TZ5QPjAQus6F9HBYjucS3S2Z0nsba13R3yWSDaY~Zqc5GoQcUm77inUsZSMClwPqLykqDE8-7XG8D4oS2PkxxKy~G2KFH6thzhCCA-U2g31i-gfoSuBVWsCUQQi0ldel5xrg__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "thumbURL2000": "https://localcdn.permanent.org/_slifty/000k-0000.thumb.w2000?t=1687549081&Expires=1687549081&Signature=UPtl9G05ubhXyIJRt6gNCZT0nbhmCzZt4UciGcJlO6PuenVL28b2vff5mJUlBVrs3~U~83uAOgp6GVRzpYx~cK4prebzVYIwv3lvIpU-p9lUlXas17mt~5RRCikZ9eMSkLCLx5QG4wdu8ncVQn3aI4zT1ErFOByZ7wiC0zp4KP25GtMCxqhibj0rdGcySdZKwkACuf-0kLH9pu2~O~PE0cCvBv~Z7GK~XDonvq3mK3ThrTW-PXFSLqPXEWk3NfB5UHoMCWFxH-Nyu4WRIxdYu~dbIR8rBflpdM7U4OytcQZQ~06FeCb7JdiGuBi1KJvr5EyZgunNO7bOnO6aTx~TgQ__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "thumbURL500": "https://localcdn.permanent.org/_slifty/000k-0000.thumb.w500?t=1687549081&Expires=1687549081&Signature=aqzIZwOMpgn~zomLe6XgTcliLwvc76NQ7vDBUpvFvoq72H5aCmv~ASXxS-Fng4ufNfU-9E9lTnOkl5MewGyHx0MmAh9fbx9SRMvNsZBAhvVJBNF-rYdKWUxwuhRu-CgWwS-aMX0kgbb4pkY5YMeCu1BqxAePzwOeTBWGSj3t17uKl8G15VRWar2f5PqUDEm9oZmx8VZmp481XMXVfl3FjjhYYwZB1VXOvATlO~VrEHGR64d0DcnnJ3C85Iu5cC9KvI32AD2KYOMKQ2iEY84Ri7tOdfyY4XjeKSQKb9v3zIZ6zs-WqF-Tc7zEVr36fkQlMqezmwbXFFhEEGZnzFFUvA__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
    "type": "type.archive.person",
    "updatedDT": "2022-06-23T19:38:01",
    "view": null,
    "viewProperty": null,
  },
]
`;
